lib/python/ensembl/brc4/runnable/process_gff3.py

```diff
@@ -636,42 +636,110 @@ def build_transcript(self, gene: SeqFeature) -> SeqFeature:
         return transcript
 
     def move_cds_to_mrna(self, gene: SeqFeature) -> SeqFeature:
-        """Move a CDS child of a gene, to the mRNA.
+        """Move CDS child features of a gene, to the mRNA.
+
+        This is to fix the case where we have the following structure:
+        gene -> [ mRNA, CDSs ]
+        and change it to 
+        gene -> [ mNRA -> [ CDSs ] ]
+        The mRNA might have exons, in which case check that they match the CDS coordinates
 
-        Raises an exception if there is not exactly 1 mRNA and 1 CDS.
+        Raises an exception if the feature structure is not recognized.
         
         Args:
-            A gene with both one mRNA and one CDS child.
+            A gene with only one transcript, to check and fix.
         Returns:
-            A transcript where the CDS child has been moved under.
+            The gene where the CDSs have been moved, if needed.
 
         """
-        
-        # This is assuming there is 1 CDS and 1 mRNA without CDS
-        cdss = []
+        # First, count the types
+        feats = []
         mrnas = []
+        cdss = []
+
+        gene_subf_clean = []
         for subf in gene.sub_features:
-            if subf.type == "CDS":
-                cdss.append(subf)
             if subf.type == "mRNA":
                 mrnas.append(subf)
+            elif subf.type == "CDS":
+                cdss.append(subf)
+            else:
+                gene_subf_clean.append(subf)
+
+        if len(cdss) > 0:
+            if len(mrnas) > 1:
+                raise Exception(f"Can't fix gene {gene.id}: contains several mRNAs and CDSs, all children of the gene")
+        else:
+            # Nothing to fix here, no CDSs to move
+            return gene
```

Comment:
Suggestion with less lines:
```suggestion
        if len(cdss) == 0:
            # Nothing to fix here, no CDSs to move
            return gene
        if len(mrnas) > 1:
            raise Exception(f"Can't fix gene {gene.id}: contains several mRNAs and CDSs, all children of the gene")
```